extended/src/main/java/io/kubernetes/client/extended/leaderelection/LeaderElector.java

```diff
@@ -15,6 +15,41 @@
 
   private static final double JITTER_FACTOR = 1.2;
 
+  public static LeaderElector newLeaderElector(LeaderElectionConfig config) {
+    if (config == null) {
+      throw new RuntimeException("Config must be provided.");
+    }
+    if (config.getLock() == null) {
+      throw new RuntimeException("Lock must be provided.");
```

Comment:
lang.IllegalArgumentException?

Reply:
definitely this and everywhere. Please don't use `RuntimeException` anywhere it's not 100% required.